### PR TITLE
ISM security

### PR DESCRIPTION
**Pull request by bowenlan-amzn**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexRunner.kt [hunk 1]
```diff
@@ -283,7 +308,9 @@ object ManagedIndexRunner : ScheduledJobRunner,
 
         if (updateResult && state != null && action != null && step != null && currentActionMetaData != null) {
             // Step null check is done in getStartingManagedIndexMetaData
-            step.preExecute(logger).execute().postExecute(logger)
+            runBlocking(InjectorContextElement(managedIndexConfig.id, settings, threadPool.threadContext, roles)) {
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexRunner.kt:311]: Don't use runBlocking

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/deletepolicy/TransportDeletePolicyAction.kt [hunk 2]
```diff
@@ -37,6 +37,8 @@ class TransportDeletePolicyAction @Inject constructor(
         val deleteRequest = DeleteRequest(IndexManagementPlugin.INDEX_MANAGEMENT_INDEX, request.policyID)
                 .setRefreshPolicy(request.refreshPolicy)
 
-        client.delete(deleteRequest, listener)
+        client.threadPool().threadContext.stashContext().use {
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/deletepolicy/TransportDeletePolicyAction.kt:40]: Can we add a comment explaining what this is doing for future reference

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/TestHelpers.kt [hunk 3]
```diff
@@ -276,10 +277,24 @@ fun randomManagedIndexConfig(
...
         policy = policy?.copy(id = ManagedIndexConfig.NO_ID, seqNo = SequenceNumbers.UNASSIGNED_SEQ_NO, primaryTerm = SequenceNumbers.UNASSIGNED_PRIMARY_TERM),
-        changePolicy = changePolicy
+        changePolicy = changePolicy,
+        user = null
+    )
+}
+
+fun randomUser(): User {
+    return User(
+        ESRestTestCase.randomAlphaOfLength(10),
```
Comment (dbbaughe) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/TestHelpers.kt:287]: Named arguments

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/addpolicy/TransportAddPolicyAction.kt [hunk 4]
```diff
@@ -73,7 +77,12 @@ class TransportAddPolicyAction @Inject constructor(
     }
 
     override fun doExecute(task: Task, request: AddPolicyRequest, listener: ActionListener<ISMStatusResponse>) {
-        AddPolicyHandler(client, listener, request).start()
+        val userStr = client.threadPool().threadContext.getTransient<String>(ConfigConstants.OPENDISTRO_SECURITY_USER_INFO_THREAD_CONTEXT)
+        user = resolveUser(User.parse(userStr))
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/addpolicy/TransportAddPolicyAction.kt:81]: What does this parse as when security plugin isn't installed?

File: build.gradle [hunk 5]
```diff
@@ -130,6 +130,8 @@ dependencies {
     compile "org.jetbrains.kotlin:kotlin-stdlib-common:${kotlin_version}"
     compile 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.7'
     compile "org.jetbrains:annotations:13.0"
+    compile "org.elasticsearch.client:elasticsearch-rest-client:${es_version}"
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:${kotlin_version}"
```
Comment (dbbaughe) [inline build.gradle:134]: What are these for

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt [hunk 6]
```diff
@@ -293,10 +289,11 @@ class ManagedIndexCoordinator(
         indexToMatchedPolicy.filterNotNullValues()
             .forEach { (index, policyID) ->
                 val indexUuid = indexMetadatas[index].indexUUID
-                if (indexUuid != null) {
-                    logger.info("index [$index] will be managed by policy [$policyID]")
+                val ismTemplate = templates[policyID]
+                if (indexUuid != null && ismTemplate != null) {
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt:293]: Do you want a warning for the else clause here?

File: src/main/resources/mappings/opendistro-ism-config.json [hunk 7]
```diff
@@ -454,6 +454,51 @@
...
+                },
+                "custom_attribute_names": {
+                  "type": "text",
+                  "fields": {
+                    "keyword": {
+                      "type": "keyword"
+                    }
+                  }
+                },
+                "user_requested_tenant": {
```
Comment (dbbaughe) [inline src/main/resources/mappings/opendistro-ism-config.json:493]: What's this?

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/ODFERestTestCase.kt [hunk 8]
```diff
@@ -106,15 +106,19 @@ abstract class ODFERestTestCase : ESRestTestCase() {
         }
     }
 
+    override fun preserveIndicesUponCompletion(): Boolean {
+        return true
+    }
+
     @Throws(IOException::class)
+    @After
```
Comment (dbbaughe) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/ODFERestTestCase.kt:114]: Why was this added back?

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestRetryFailedManagedIndexActionIT.kt [hunk 9]
```diff
@@ -103,77 +103,77 @@ class RestRetryFailedManagedIndexActionIT : IndexStateManagementRestTestCase() {
...
-                mapOf(
-                    "index_name" to indexName,
-                    "index_uuid" to getUuid(indexName),
-                    "reason" to "This index is not being managed."
-                )
-            )
-        )
-        assertAffectedIndicesResponseIsEqual(expectedErrorMessage, actualMessage)
-    }
+    // fun `test index not being managed`() {
```
Comment (dbbaughe) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestRetryFailedManagedIndexActionIT.kt:135]: Why commented out

Comment (bowenlan-amzn) [reply src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestRetryFailedManagedIndexActionIT.kt:135]: uncomment

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/ODFERestTestCase.kt (hunk 8 above)
Comment (bowenlan-amzn) [reply src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/ODFERestTestCase.kt:114]: when I am running test against a cluster with security plugin, if just remove all indices, it will cause problem. We need to keep security system index.

File: src/main/resources/mappings/opendistro-ism-config.json (hunk 7 above)
Comment (bowenlan-amzn) [reply src/main/resources/mappings/opendistro-ism-config.json:493]: actually I can remove this, this part should be part of User object. But since we don't use this, and cannot see any reason use this in future, so in the code, we only keep the other four fields.

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/addpolicy/TransportAddPolicyAction.kt (hunk 4 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/addpolicy/TransportAddPolicyAction.kt:81]: I remember user saved is null when manual test

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/TestHelpers.kt (hunk 3 above)
Comment (bowenlan-amzn) [reply src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/TestHelpers.kt:287]: image
suprise

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/deletepolicy/TransportDeletePolicyAction.kt (hunk 2 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/deletepolicy/TransportDeletePolicyAction.kt:40]: copied one from alerting to our AddPolicyAction
```
/*
 * Remove security context before you call elasticsearch api's. By this time, permissions required
 * to call this api are validated.
 * Once system-indices [https://github.com/opendistro-for-elasticsearch/security/issues/666] is done, we
 * might further improve this logic. Also change try to kotlin-use for auto-closable.
 */
```

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/getpolicy/TransportGetPolicyAction.kt [hunk 10]
```diff
@@ -35,6 +36,8 @@ import org.elasticsearch.rest.RestStatus
 import org.elasticsearch.tasks.Task
 import org.elasticsearch.transport.TransportService
 
+private val log = LogManager.getLogger(TransportGetPolicyAction::class.java)
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/getpolicy/TransportGetPolicyAction.kt:39]: Are you using this?

Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/getpolicy/TransportGetPolicyAction.kt:39]: removed

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt (hunk 6 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt:293]: add a warning log:
```
else {
    logger.warn("index [$index] with null index uuid [$indexUuid] or " +
        "matching template [$ismTemplate] for this index is null.")
}
```

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt [hunk 11]
```diff
@@ -59,14 +63,23 @@ class TransportIndexPolicyAction @Inject constructor(
 ) : HandledTransportAction<IndexPolicyRequest, IndexPolicyResponse>(
         IndexPolicyAction.NAME, transportService, actionFilters, ::IndexPolicyRequest
 ) {
+
+    @Volatile lateinit var policy: Policy
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt:67]: Move this inside doExecute or it'll leak across requests

Comment (dbbaughe) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt:67]: Or just move onto the IndexPolicyHandler class

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestDeletePolicyAction.kt [hunk 12]
```diff
@@ -40,6 +43,8 @@ class RestDeletePolicyAction : BaseRestHandler() {
 
     @Throws(IOException::class)
     override fun prepareRequest(request: RestRequest, client: NodeClient): RestChannelConsumer {
+        log.info("${request.method()} ${request.path()}")
```
[the remark this answers is not in the document]
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestDeletePolicyAction.kt:46]: removed

File: build.gradle (hunk 5 above)
Comment (bowenlan-amzn) [reply build.gradle:134]: kotlin-stdlib-jdk8 is for this line `client.threadPool().threadContext.stashContext().use {...}`

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/addpolicy/TransportAddPolicyAction.kt (hunk 4 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/addpolicy/TransportAddPolicyAction.kt]: userStr is null
parsed user is `parsed user: User[name=, backend_roles=[], roles=[], custom_attribute_names=[], user_requested_tenant=null]`

in the managed-index document:
```
"user": {
  "name": "",
  "backend_roles": [],
  "roles": [],
  "custom_attribute_names": [],
  "user_requested_tenant": null
}
```

File: build.gradle (hunk 5 above)
Comment (bowenlan-amzn) [reply build.gradle:134]: removed rest-client dependency

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestDeletePolicyAction.kt [hunk 13]
```diff
@@ -28,6 +29,8 @@ import org.elasticsearch.rest.RestRequest.Method.DELETE
 import org.elasticsearch.rest.action.RestStatusToXContentListener
 import java.io.IOException
 
+private val log = LogManager.getLogger(RestDeletePolicyAction::class.java)
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestDeletePolicyAction.kt:32]: Remove

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/util/ManagedIndexUtilsTests.kt [hunk 14]
```diff
@@ -90,52 +93,52 @@ class ManagedIndexUtilsTests : ESTestCase() {
...
-        assertTrue("Incorrect request type", request is IndexRequest)
-        val source = (request as IndexRequest).source()
-        val managedIndexConfig = contentParser(source).parseWithType(parse = ManagedIndexConfig.Companion::parse)
-
-        assertEquals("Incorrect index on ManagedIndexConfig source", clusterConfigToCreate.index, managedIndexConfig.index)
-        assertEquals("Incorrect name on ManagedIndexConfig source", clusterConfigToCreate.index, managedIndexConfig.name)
-        assertEquals("Incorrect index uuid on ManagedIndexConfig source", clusterConfigToCreate.uuid, managedIndexConfig.indexUuid)
-        assertEquals("Incorrect policy_id on ManagedIndexConfig source", clusterConfigToCreate.policyID, managedIndexConfig.policyID)
-    }
+    // fun `test get create managed index requests`() {
```
Comment (dbbaughe) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/util/ManagedIndexUtilsTests.kt:96]: Why commented out? If not relevant anymore delete.

File: build.gradle [hunk 15]
```diff
@@ -130,6 +130,7 @@ dependencies {
     compile "org.jetbrains.kotlin:kotlin-stdlib-common:${kotlin_version}"
     compile 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.7'
     compile "org.jetbrains:annotations:13.0"
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:${kotlin_version}"
```
Comment (dbbaughe) [inline build.gradle:133]: Seems like you mentioned this is just for one method. Possible to pull in that method ourselves instead of adding a whole dependency?

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt [hunk 16]
```diff
@@ -117,14 +134,14 @@ class TransportIndexPolicyAction @Inject constructor(
                     val conflictingPolicyTemplates = policyToTemplateMap.findConflictingPolicyTemplates(request.policyID, indexPatterns, priority)
                     if (conflictingPolicyTemplates.isNotEmpty()) {
                         val errorMessage = "new policy ${request.policyID} has an ism template with index pattern $indexPatterns " +
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt:136]: Capitalize "**N**ew" and ISM

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/ODFERestTestCase.kt (hunk 8 above)
Comment (dbbaughe) [reply src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/ODFERestTestCase.kt:114]: Yes, but I meant the tests were rewritten to assume indices were *not* being deleted anymore. So we do not actually want this to run @After every test

Comment (dbbaughe) [reply src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/ODFERestTestCase.kt:114]: Whoops, sorry Maxim 😬

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt (hunk 11 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt]: moved to IndexPolicyHandler class

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestDeletePolicyAction.kt (hunk 13 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestDeletePolicyAction.kt:32]: removed

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/util/ManagedIndexUtilsTests.kt (hunk 14 above)
Comment (bowenlan-amzn) [reply src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/util/ManagedIndexUtilsTests.kt:96]: removed

File: build.gradle (hunk 15 above)
Comment (bowenlan-amzn) [reply build.gradle:133]: make sense, copied that method from kotlin stdlib to our RestHandlerUtils

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt (hunk 16 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt:136]: addressed

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/ODFERestTestCase.kt (hunk 8 above)
Comment (bowenlan-amzn) [reply src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/ODFERestTestCase.kt]: move these to only ISM security test file

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/removepolicy/TransportRemovePolicyAction.kt [hunk 17]
```diff
@@ -55,7 +56,9 @@ class TransportRemovePolicyAction @Inject constructor(
         RemovePolicyAction.NAME, transportService, actionFilters, ::RemovePolicyRequest
 ) {
     override fun doExecute(task: Task, request: RemovePolicyRequest, listener: ActionListener<ISMStatusResponse>) {
-        RemovePolicyHandler(client, listener, request).start()
+        client.threadPool().threadContext.stashContext().use {
```
Comment (qreshi) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/removepolicy/TransportRemovePolicyAction.kt:59]: I see that the `TransportAddPolicyAction` is retrieving user and passing it to the `AddPolicyHandler` but we don't seem to be adding user context here

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexRunner.kt (hunk 1 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexRunner.kt]: use withContext and do roleInject.close at the end of block.
@skkosuri-amzn alerting can change to this also

**Review by qreshi — COMMENTED** [review]
I see that some transport classes don't have `user` being pulled in and used.

Just to confirm, will we not be filtering things like policies (TransportGetPolicy) and managed jobs (TransportExplain) based on users? Not sure what the recommended behavior is in multi-tentant configurations with security enabled and whether or not we want to filter out those things users don't have permissions for.

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/addpolicy/TransportAddPolicyAction.kt [hunk 18]
```diff
@@ -73,13 +77,25 @@ class TransportAddPolicyAction @Inject constructor(
...
 
     override fun doExecute(task: Task, request: AddPolicyRequest, listener: ActionListener<ISMStatusResponse>) {
-        AddPolicyHandler(client, listener, request).start()
+        val userStr = client.threadPool().threadContext.getTransient<String>(ConfigConstants.OPENDISTRO_SECURITY_USER_INFO_THREAD_CONTEXT)
+        val user = resolveUser(User.parse(userStr))
+
+        /*
+         * Remove security context before you call elasticsearch api's. By this time, permissions required
+         * to call this api are validated.
+         * Once system-indices [https://github.com/opendistro-for-elasticsearch/security/issues/666] is done, we
```
[the remark this answers is not in the document]
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/addpolicy/TransportAddPolicyAction.kt:86]: remove these 2 lines of comment, we are using the kotlint use now

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexRunner.kt [hunk 19]
```diff
@@ -197,6 +205,22 @@ object ManagedIndexRunner : ScheduledJobRunner,
             return
         }
 
+        /*
+         * We need to handle 3 cases:
+         * 1. ISM jobs created by older versions and never updated. These job wont have User details in the
+         * job object. `managedIndexConfig.user` will be null. Insert `all_access, AmazonES_all_access` role.
+         * 2. ISM jobs are created when security plugin is disabled, these will have empty User object.
```
[the remark this answers is not in the document]
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexRunner.kt:213]: yes, addressed

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt [hunk 20]
```diff
@@ -293,11 +289,15 @@ class ManagedIndexCoordinator(
...
+                    logger.info("index [$index] will be managed by policy [$policyID] of roles [${ismTemplate.user?.roles}]")
                     updateManagedIndexReqs.add(
-                        managedIndexConfigIndexRequest(index, indexUuid, policyID, jobInterval)
+                        managedIndexConfigIndexRequest(index, indexUuid, policyID, jobInterval, ismTemplate.user)
                     )
+                } else {
+                    logger.warn("index [$index] with null index uuid [$indexUuid] or " +
```
[the remark this answers is not in the document]
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt:299]: Thanks, changed

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/removepolicy/TransportRemovePolicyAction.kt (hunk 17 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/removepolicy/TransportRemovePolicyAction.kt:59]: when we retrieve user, we are trying to save it to managed-index. So we only retrieve user for addPolicy, changePolicy, indexPolicy(save to ismTemplate field). Other actions we don't do that.

**Review by dbbaughe — APPROVED** [review]

**Review by qreshi — APPROVED** [review]
